spec.emu

```diff
@@ -12,162 +12,113 @@ copyright: false
-        <h1>%NumericRangeIteratorPrototype%.[@@toStringTag]</h1>
-        <p>The initial value of the @@toStringTag property is the String value `"NumericRangeIterator"`.</p>
-        <p>This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</p>
-    </emu-clause>
-    <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-start">
-        <h1>%NumericRangeIteratorPrototype%.start</h1>
-        <p>%NumericRangeIteratorPrototype%.start is an accessor property with attributes { [[Enumerable]]: false, [[Configurable]]: *true*}. The [[Get]] attribute is defined as follows:</p>
-        <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-start-getter">
-            <h1>get %NumericRangeIteratorPrototype%.start</h1>
-            <emu-alg>
-                1. Perform ? RequireInternalSlot(*this*, [[start]]).
-                1. Return *this*.[[start]].
-            </emu-alg>
-        </emu-clause>
-    </emu-clause>
-    <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-end">
-        <h1>%NumericRangeIteratorPrototype%.end</h1>
-        <p>%NumericRangeIteratorPrototype%.end is an accessor property with attributes { [[Enumerable]]: false, [[Configurable]]: *true*}. The [[Get]] attribute is defined as follows:</p>
-        <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-end-getter">
-            <h1>get %NumericRangeIteratorPrototype%.end</h1>
-            <emu-alg>
-                1. Perform ? RequireInternalSlot(*this*, [[end]]).
-                1. Return *this*.[[end]].
-            </emu-alg>
-        </emu-clause>
-    </emu-clause>
-    <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-inclusive">
-        <h1>%NumericRangeIteratorPrototype%.inclusive</h1>
-        <p>%NumericRangeIteratorPrototype%.inclusive is an accessor property with attributes { [[Enumerable]]: false, [[Configurable]]: *true*}. The [[Get]] attribute is defined as follows:</p>
-        <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-inclusive-getter">
-            <h1>get %NumericRangeIteratorPrototype%.isEnclusiveEnd</h1>
-            <emu-alg>
-                1. Perform ? RequireInternalSlot(*this*, [[inclusiveEnd]]).
-                1. Return *this*.[[inclusiveEnd]].
-            </emu-alg>
+    <emu-clause id="sec-iteration" number="1">
+        <h1>Iteration</h1>
+
+        <emu-clause id="sec-properties-of-the-iterator-constructor" number="2">
+            <h1>Properties of the Iterator Constructor</h1>
+
+            <emu-clause id="sec-iterator.range">
+                <h1>Iterator.range ( _start_, _end_, _option_ )</h1>
+                <emu-alg>
+                    1. If _start_ is a Number, return ? CreateNumericRangeIterator(_start_, _end_, _option_, ~number-range~).
```

Comment:
Shouldn’t this coerce start and end to a primitive first, like every other builtin api?

Reply:
if we coerce it to a primitive, we will lose the possibility to add more overload in the future, for example, supporting a `Date` range.

```js
// for example:
Iterator.range(new Date('Jan 1 2020'), new Date(), { step: '1 day' })
```

Reply:
That seems like something more appropriate to put on Date (or better, on Temporal) than in this function.